components/kyma-environment-broker/internal/process/operation_manager.go

```diff
@@ -0,0 +1,110 @@
+	return om.update(operation, domain.Succeeded, description, log)
+}
+
+// OperationFailed marks the operation as failed and returns status of the operation's update
+func (om *OperationManager) OperationFailed(operation internal.Operation, description string, err error, log logrus.FieldLogger) (internal.Operation, time.Duration, error) {
+	return om.update(operation, domain.Failed, description, log)
+}
+
+// OperationCanceled marks the operation as canceled and returns status of the operation's update
+func (om *OperationManager) OperationCanceled(operation internal.Operation, description string, log logrus.FieldLogger) (internal.Operation, time.Duration, error) {
+	return om.update(operation, orchestration.Canceled, description, log)
+}
+
+// RetryOperation checks if operation should be retried or if it's the status should be marked as failed
+func (om *OperationManager) RetryOperation(operation internal.Operation, errorMessage string, err error, retryInterval time.Duration, maxTime time.Duration, log logrus.FieldLogger) (internal.Operation, time.Duration, error) {
+	log.Infof("Retry Operation was triggered with message: %s", errorMessage)
+	log.Infof("Retrying for %s in %s steps", maxTime.String(), retryInterval.String())
+	if time.Since(operation.UpdatedAt) < maxTime {
+		return operation, retryInterval, nil
+	}
+	log.Errorf("Aborting after %s of failing retries", maxTime.String())
+	op, retry, err := om.OperationFailed(operation, errorMessage, err, log)
+	if err == nil {
+		err = fmt.Errorf("Too many retries")
+	} else {
+		err = fmt.Errorf("Failed to set status for operation after too many retries: %v", err)
+	}
+	return op, retry, err
+}
+
+// RetryOperationWithoutFail checks if operation should be retried or updates the status to InProgress, but omits setting the operation to failed if maxTime is reached
+func (om *OperationManager) RetryOperationWithoutFail(operation internal.Operation, description string, retryInterval, maxTime time.Duration, log logrus.FieldLogger) (internal.Operation, time.Duration, error) {
+	log.Infof("Retry Operation was triggered with message: %s", description)
+	log.Infof("Retrying for %s in %s steps", maxTime.String(), retryInterval.String())
+	if time.Since(operation.UpdatedAt) < maxTime {
+		return operation, retryInterval, nil
+	}
+	// update description to track failed steps
+	op, repeat, err := om.update(operation, domain.InProgress, description, log)
+	if repeat != 0 {
+		return op, repeat, err
+	}
+
+	log.Errorf("Omitting after %s of failing retries", maxTime.String())
+	return op, 0, nil
+}
+
+// RetryOperationOnce retries the operation once and fails the operation when call second time
+func (om *OperationManager) RetryOperationOnce(operation internal.Operation, errorMessage string, err error, wait time.Duration, log logrus.FieldLogger) (internal.Operation, time.Duration, error) {
+	return om.RetryOperation(operation, errorMessage, err, wait, wait+1, log)
```

Comment:
time.Duration unit is nanosecond so this wait+1 is puzzling me.

Reply:
it was present in all of the other managers, and I think it's just to ensure it retries exactly once :)